### PR TITLE
Added transform_bindings to common.py

### PR DESCRIPTION
**Pull request by ajinkya1986**

### CODE REVIEW

File: cartography/intel/gcp/util/common.py [hunk 1]
```diff
@@ -19,3 +20,57 @@ def parse_and_validate_gcp_requested_syncs(gcp_requested_syncs: str) -> List[str
                 f'Our full list of valid values is: {valid_syncs}.',
             )
     return validated_resources
+
+def transform_bindings(bindings: Dict, project_id: str) -> tuple:
```
Comment (mpurusottamc) [inline cartography/intel/gcp/util/common.py:24]: let's rename this function to make it specific like `transform_iam_bindings`.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
one minor change needed.

File: cartography/intel/gcp/util/common.py [hunk 2]
```diff
@@ -19,3 +20,57 @@ def parse_and_validate_gcp_requested_syncs(gcp_requested_syncs: str) -> List[str
                 f'Our full list of valid values is: {valid_syncs}.',
             )
     return validated_resources
+
+def transform_iam_bindings(bindings: Dict, project_id: str) -> tuple:
```
Comment (mpurusottamc) [inline cartography/intel/gcp/util/common.py:24]: There are 2 use cases missing in this code.
1. code in iam load_bindings has changed drastically. Can you please compare, if we need to make any changes here?
2. deleted users, serviceAccounts, groups, etc. is handled in iam, we should ignore those entities

File: cartography/intel/gcp/util/common.py [hunk 3]
```diff
@@ -19,3 +20,57 @@ def parse_and_validate_gcp_requested_syncs(gcp_requested_syncs: str) -> List[str
...
+                        domains.append({
+                            "id": f'projects/{project_id}/domains/{dmn}',
+                            "email": dmn,
+                            "name": dmn,
+                        })
+
+                    elif member.startswith('serviceAccount:'):
+                        sac = member[len('serviceAccount:'):]
+                        service_account.append({
+                            "id": f'projects/{project_id}/service_account/{sac}',
```
Comment (mpurusottamc) [inline cartography/intel/gcp/util/common.py:64]: this id format is incorrect.

File: cartography/intel/gcp/apigateway.py [hunk 4]
```diff
@@ -367,7 +367,7 @@ def get_api_gateway_policy_entities(apigateway: Resource, gateway: Dict, project
         iam_policy = apigateway.projects().locations().gateways().getIamPolicy(
             resource=gateway.get('name')).execute()
         bindings = iam_policy.get('bindings', [])
-        entity_list, public_access = iam.transform_bindings(bindings, project_id)
+        entity_list, public_access = transform_iam_bindings(bindings, project_id)
```
Comment (mpurusottamc) [inline cartography/intel/gcp/apigateway.py:370]: I don't see this output used anywhere. `get_api_gateway_policy_entities` method is not used at all. let's remove this method if not used?

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
minor changes needed. 

Also, what would be additional performance impact with new api calls? 
We are calling getIamPolicy for each resource. That would add to the total time it takes to run cartography.

File: cartography/intel/gcp/util/common.py (hunk 2 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/util/common.py:24]: Added in common.py file to ingnore deleted entities

File: cartography/intel/gcp/apigateway.py (hunk 4 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/apigateway.py:370]: It is now used in the `get_gateways` method

File: cartography/intel/gcp/util/common.py [hunk 5]
```diff
@@ -61,10 +61,12 @@ def transform_iam_bindings(bindings: Dict, project_id: str) -> tuple:
                     elif member.startswith('serviceAccount:'):
                         sac = member[len('serviceAccount:'):]
                         service_account.append({
-                            "id": f'projects/{project_id}/service_account/{sac}',
+                            "id": f'projects/{project_id}/serviceAccounts/{sac}',
                             "email": sac,
                             "name": sac,
                         })
+                    elif member.startswith('deleted:'):
+                        pass
```
Comment (mpurusottamc) [inline cartography/intel/gcp/util/common.py:69]: should this be pass or continue?

**Review by mpurusottamc — COMMENTED** [review]
need clarification.

File: cartography/intel/gcp/util/common.py (hunk 5 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/util/common.py:69]: yes it should be `continue`

**Review by mpurusottamc — APPROVED** [review]
looks good.

**Pull request closed without merging**